docs/manual/guides/formular-speichern.de.md

```diff
@@ -0,0 +1,550 @@
+      <?php if ($this->maxFileSize): ?>
+        <input type="hidden" name="MAX_FILE_SIZE" value="<?= $this->maxFileSize ?>">
+      <?php endif; ?>
+    <?php endif; ?>
+    <?= $this->hidden ?>
+    <?= $this->fields ?>
+  </div>
+</form>
+
+</div>
+<!-- indexer::continue -->
+
+<script src="files\EDITPATH\pickadate\picker.js"></script>
+<script src="files\EDITPATH\picker.date.js"></script>
+<script src="files\EDITPATH\picker.time.js"></script>
+<script src="files\EDITPATH\translations\de_DE.js"></script>
+
+<script>
+$(document).ready(function(){
+	
+	// create javascript array wirh dates to disable and consider php/js difference
+	var arrDisableDate = [
+
+	<?php	
+	$leadsData = $dataCollector-> getExportData(); 
+	
+	foreach($leadsData as $row) {
+		foreach($row as $content) {
+		    if ( $content['name'] == "startDate" )  {
+				$startTimestamp = $content['value'];
+				$startTimestampJs = date('Y,m,d', strtotime('-1 month', $startTimestamp));
+			}
+			
+			if ( $content['name'] == "endDate" && !empty(trim($content['value'])) )  {
+				$endTimestamp = $content['value'];
+				$endTimestampJs = date('Y,m,d', strtotime('-1 month', $endTimestamp));	
+			} else {
+				$endTimestamp = null;
+			}
+		}
+	
+		// check for optional endDate value
+		if ( !empty(trim($endTimestamp)) ) {
+			print "{from:[". $startTimestampJs ."], to:[". $endTimestampJs ."]},"; 
+		} else {
+	  		print "[". $startTimestampJs ."],"; 
+		}
+	}
+	?>	
+	]
+	
+    $.extend($.fn.pickadate.defaults, {
+        showMonthsShort: true,
+        showWeekdaysFull: false,        
+        labelMonthNext: 'nächster Monat',
+        labelMonthPrev: 'vorheriger Monat',
+        closeOnSelect: true,
+        closeOnClear: true,
+        format: 'dd.mm.yyyy',
+        formatSubmit: 'dd.mm.yyyy',
+        hiddenName: true,
+        min: new Date(),
+        disable: arrDisableDate
+    })    
+
+    var from_$input = $('input.js_startDate').pickadate(),
+        from_picker = from_$input.pickadate('picker')
+    
+    var to_$input = $('input.js_endDate').pickadate(),
+        to_picker = to_$input.pickadate('picker')
+
+    // Check if there’s a “from” or “to” date to start with.
+    if ( from_picker.get('value') ) {
+      to_picker.set('min', from_picker.get('select'))
+    }
+    if ( to_picker.get('value') ) {
+      from_picker.set('max', to_picker.get('select'))
+    }
+    
+    // When something is selected, update the “from” and “to” limits.
+    from_picker.on('set', function(event) {
+      if ( event.select ) {
+        to_picker.set('min', from_picker.get('select'))    
+      }
+      else if ( 'clear' in event ) {
+        to_picker.set('min', false)
+      }
+    })
+    
+    to_picker.on('set', function(event) {
+      if ( event.select ) {
+        from_picker.set('max', to_picker.get('select'))
+      }
+      else if ( 'clear' in event ) {
+        from_picker.set('max', false)
+      }
+    })    
+}) 
+</script>
+```
```

Comment:
Instead of introducing PHP and JavaScript Code directly in a template, it would be much cleaner to implement your own form field. I am not sure we should promote writing PHP code like this directly in templates.

Reply:
> ...it would be much cleaner to implement your own form field. 
What exactly do you mean by that?
A conversion via a separate extension? But that would probably be too extensive for normal users (like me).

Reply:
You do not need to create an extension to implement a form field.

This is how you implement your own form field, i.e. front end widget: https://docs.contao.org/dev/framework/widgets/#creating-front-end-widgets

You can extend from `FormTextField` and override the `parse` method to add the necessary data to the widget as well as integrate JavaScript via `$GLOBALS`.

>  But that would probably be too extensive for normal users (like me).

What is a normal user? Regardless of the type of user, there are still things that we should not officially promote.

Reply:
I will try ...